api/python/provisioner/errors.py

```diff
@@ -286,3 +286,99 @@ def __repr__(self):
             "{}(reason={!r})"
             .format(self.__class__.__name__, self.reason)
         )
+
+
+class HareError(ProvisionerError):
+    _prvsnr_type_ = True
+
+    def __init__(self, funct_name: str, reason: Union[Exception, str]):
+        self.funct_name = funct_name
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            'Hare Cluster not in healthy state, reason: {!r}'
+            .format(self.reason)
+        )
+
+class HareClusterError(HareError):
+    pass
+
+class EnsureUpdateRepoConfigError(ProvisionerError):
+    _prvsnr_type_ = True
+
+    def __init__(self, reason: Union[Exception, str]):
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            'Repos are not configured or yum package error, reason: {!r}'
+            .format(self.reason)
+        )
+
+
+class UpdateComponentError(ProvisionerError):
+    _prvsnr_type_ = True
+
+    def __init__(self, reason: Union[Exception, str]):
```

Comment:
that exception is quite specific since it uses `Component` word in the name, please add `component` attibute and updated related code